app/services/foreman/renderer/scope/macros/base.rb

```diff
@@ -117,15 +117,21 @@ def pxe_kernel_options
             example "save_to_file(shell_escape('/tmp/a file with spaces'), nil) # => 'cp /dev/null /tmp/a\ file\ with\ spaces'"
           end
           def save_to_file(filename, content, verbatim: false)
-            delimiter = 'EOF-' + Digest::SHA512.hexdigest(filename)[0..7]
             if content.empty?
               "cp /dev/null #{filename}"
             elsif verbatim
-              content = Base64.encode64(content)
-              "cat << #{delimiter} | base64 -d > #{filename}\n#{content}#{delimiter}"
+              base64 = Base64.encode64(content)
+              content_assignments = base64.split("\n").map { |l| "echo #{Shellwords.shellescape(l)}" }.join("\n")
```

Comment:
Is the escaping even needed? Base64 shouldn't contain any problematic characters.

Reply:
@ShimShtein ping